### PR TITLE
Add documentation for the nosupertype annotation

### LINKED ISSUE #492: `nosupertype` annotation is missing from tutorial
See https://github.com/ponylang/rfcs/blob/main/text/0054-subtyping-exclusion.md for information needed to add.

### PR DESCRIPTION
**Pull request by SeanTAllen**
Closes #492

### CODE REVIEW

File: docs/appendices/annotations.md [hunk 1]
```diff
@@ -89,3 +89,35 @@ class \nodoc\Foo
...
+    | let a': A => None
+    end
+```
+
+The above code won't compile because you could supply `Empty ref`. Doing so results in a compiler error about an unsafe match because we would need to distinguish between `Empty val` and `Empty ref` at runtime.
+
+By adding `nosupertype` to the definition of `Empty`, we declare that `Empty` is not a subtype of `Any` and thereby allow the code to compile as there is no longer an unsafe match.
+
+```pony
+class \nosuperclass\ Empty
```
Comment (jemc) [inline docs/appendices/annotations.md:114]: Wrong annotation name?
```suggestion
class \nosupertype\ Empty
```

File: docs/appendices/annotations.md [hunk 2]
```diff
@@ -89,3 +89,35 @@ class \nodoc\Foo
...
+
+class Foo
+  fun foo[A: Any](a: (A | Empty val)) =>
+    match consume a
+    | let a': A => None
+    end
+```
+
+`nosuperclass` is particularly valuable when constructing generic classes like collections that need a marker class to describe "lack of an item".
```
Comment (jemc) [inline docs/appendices/annotations.md:123]: Wrong annotation name?
```suggestion
`nosupertype` is particularly valuable when constructing generic classes like collections that need a marker class to describe "lack of an item".
```

**Review by jemc — COMMENTED** [review]
I totally forgot this feature even existed.

I don't even remember discussing, voting on, or approving the RFC.

**Comment by SeanTAllen** [comment]
It was part of a 2 parter from Benoit. Only half got implemented.

File: docs/appendices/annotations.md [hunk 3]
```diff
@@ -89,3 +89,35 @@ class \nodoc\Foo
    We don't want this class and its methods to appear in generated documentation
    """
 ```
+
+### `nosupertype`
+
+Recognised on objects(`actor`, `class`, `primitive`, `struct`). A type annotated with `nosupertype` will not be a subtype of any other type (except _), even if the type structurally provides an interface. If a `nosupertype` type has a provides list, a compiler error is reported. As a result, a `nosupertype` type is excluded from both nominal and structural subtyping.
```
Comment (ergl) [inline docs/appendices/annotations.md:95]: What's the meaning of `_` in this context?

Comment (SeanTAllen) [reply docs/appendices/annotations.md:95]: I lifted that directly from the RFC. I need to go look and see again what specifically `_` is in the type system.

**Pull request merged**